Polygon.Domain/ApiModels/PolygonApiModel.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Text;
+
+namespace PolygonMap.Domain.ApiModels
+{
+    public class PolygonApiModel
+    {
+
+       
+        public int PolygonID { get; set; }
+        [Required]
+        public int ShapeID { get; set; }
```

Comment:
This is detached from Shape table in database level. Why have you cut the relation between the tables and instead you've used raw ids? This will disable deletion protection, cascading or set_null behaviors. Also it will disable the validation that this id exists in the other table

Reply:
This field should be deleted and use the ShapeApiModel field instead

Reply:
Please check out [this documentionat](https://docs.microsoft.com/en-us/aspnet/mvc/overview/getting-started/getting-started-with-ef-using-mvc/creating-a-more-complex-data-model-for-an-asp-net-mvc-application)